tools/setup/emoji/build_emoji

```diff
@@ -349,6 +383,22 @@ def generate_map_files(cache_path: str, emoji_catalog: Dict[str, List[str]]) ->
             )
         )
 
+    # This is the more official API for mobile to fetch data about emoji.
+    # emoji_codes.json has a lot of goo, and we're creating this new file as a cleaner data format to move towards.
+    # We could add the rest of the data into this API-described data, and then the web client could switch to that
+    # which would allow us to drop the existing file. But we'll probably instead do #18121 which will make this file
+    # obsolete. So this is a temporary solution. CZO discussion:
+    # https://chat.zulip.org/#narrow/stream/378-api-design/topic/currently.20supported.20emoji/near/1394598
+    EMOJI_API_FILE_PATH = os.path.join(cache_path, "emoji_api.json")
+    with open(EMOJI_API_FILE_PATH, "wb") as emoji_api_file:
+        emoji_api_file.write(
+            orjson.dumps(
+                {
+                    "name_to_codepoint": name_to_codepoint,
```

Comment:
Should be `name_to_code` (or some other name we pick):
https://chat.zulip.org/#narrow/stream/378-api-design/topic/currently.20supported.20emoji/near/1385346

(`name_to_codepoint` is an inaccurate name, because each value can represent a whole sequence of codepoints, not just one. Plus, they're encoded in [an idiosyncratic format](https://zulip.com/api/add-reaction#parameter-reaction_type):
> `unicode_emoji` : In this namespace, `emoji_code` will be a dash-separated hex encoding of the sequence of Unicode codepoints that define this emoji in the Unicode specification.

so the only way to really make sense of them is by reference to the `emoji_code` concept in the Zulip API.)

Reply:
I copied this from `emoji_codes.json` using the same key for the same data, but that makes sense!

Reply:
Yep, this is specifically something I wanted to fix relative to what's in that file :slightly_smiling_face: